vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc

```diff
@@ -954,21 +959,32 @@ bool GRANT::loadFromJson(const std::string & json) {
 
   if (!error) {
     promotionId = d["promotionId"].GetString();
+
+    if (d.HasMember("type") && d["type"].IsString()){
+      type = d["type"].GetString();
+    }
     return !error;
   }
 
   // On successful grant
   error = !(
       d.HasMember("altcurrency") && d["altcurrency"].IsString() &&
       d.HasMember("expiryTime") && d["expiryTime"].IsNumber() &&
-      d.HasMember("probi") && d["probi"].IsString() &&
-      d.HasMember("type") && d["type"].IsString());
+      d.HasMember("probi") && d["probi"].IsString());
 
   if (!error) {
     altcurrency = d["altcurrency"].GetString();
     expiryTime = d["expiryTime"].GetUint64();
     probi = d["probi"].GetString();
-    type = d["type"].GetString();
+    if (d.HasMember("type") && d["type"].IsString()) {
+      type = d["type"].GetString();
+    }
+
+#if defined(OS_ANDROID)
+    if (type == "ugp") {
```

Comment:
Please decouple into platform specific classes

Reply:
Issue created for that https://github.com/brave/brave-browser/issues/6200